File: app/src/main/java/com/minerdev/greformanager/GeocodeResult.java
package com.minerdev.greformanager;

import java.util.ArrayList;

/**
* y는 latitude, 즉 위도,
* x는 longitude, 즉 경도.
* **/
public class GeocodeResult {
    public String status;
    public Meta meta;
    public ArrayList<Address> addresses;
    public String errorMessage;

    public class Meta {
        public Integer totalCount;
        public Integer page;
        public Integer count;
    }

    public class Address {
        public String roadAddress;
        public String jibunAddress;
        public String englishAddress;
        public ArrayList<AddressElement> addressElements;
        public String x; // latitude
        public String y; // longitude
        public Double distance;
    }

    public class AddressElement {
        public ArrayList<String> types;
        public String longName;
        public String shortName;
        public String code;
    }
}

File: app/src/main/java/com/minerdev/greformanager/GeocodeResult.kt
package com.minerdev.greformanager

import java.util.*

/**
 * y는 latitude, 즉 위도,
 * x는 longitude, 즉 경도.
 */
class GeocodeResult {
    var status: String = ""
    var meta: Meta = Meta()
    var addresses: ArrayList<Address>? = null
    var errorMessage: String = ""

    class Meta {
        var totalCount: Int = 0
        var page: Int = 0
        var count: Int = 0
    }

    class Address {
        var roadAddress: String = ""
        var jibunAddress: String = ""
        var englishAddress: String = ""
        var addressElements: ArrayList<AddressElement>? = null
        var x // latitude
                : String = ""
        var y // longitude
                : String = ""
        var distance: Double = 0.0
    }

    class AddressElement {
        var types: ArrayList<String>? = null
        var longName: String = ""
        var shortName: String = ""
        var code: String = ""
    }
}